### PR TITLE
Add requireInteraction flag to notifications

### LINKED ISSUE #52: Are notifications expected to remain on the user's screen on desktop?
Chrome currently keeps all notifications on the screen indefinitely (for desktop).

Most notifications can be moved to an off-screen notification center automatically. We've received feedback that some major players use setTimeout() in a Service Worker to automatically remove a notification from the user's screen after a number of seconds. This is a huge hack and only works because of Chrome's Service Worker-lifetime heuristics :/.

However, some notifications can't. Consider a calendar which may choose for a notification to remain on-screen until the event has finished. Other notifications that require the user to act may also prefer a more persistent display.

We'd like to start auto-closing them, as the vast majority of users and developers prefer this, but would like an option for developers to opt-in to keeping them on the screen if they deem this important.

What about an `autominimize` property, a boolean defaulting to `true`? The name certainly needs some bikeshedding. I'd prefer not to have the minimize-time specified by developers to avoid breaking user expectations.

+@sicking

### PR DESCRIPTION
**Pull request by johnmellor**
Resolves #52

### CODE REVIEW

**Comment by johnmellor** [comment]
(I haven't run bikeshed yet, because when I do so it reformats the entire html file, most of which are whitespace changes, but in particular it adds the following spurious section. I could remove it, but it seems we should track down why it's being added since it's redundant with our !Participate links)

    +     <dt>Issue Tracking:
    +     <dd><a href="https://github.com/johnmellor/notifications/issues/">GitHub</a>

**Comment by johnmellor** [comment]
Ok, I've run bikeshed now (and filed https://github.com/tabatkins/bikeshed/issues/470).

[HTML diff](http://services.w3.org/htmldiff?doc1=https%3A%2F%2Frawgit.com%2Fjohnmellor%2Fnotifications%2F91bbe10d4030bbf71c8624c7078778be9092d0ff%2Fnotifications.html&doc2=https%3A%2F%2Frawgit.com%2Fjohnmellor%2Fnotifications%2F54cd9830294906a0217e8193be158cd654338aef%2Fnotifications.html) in case it helps review.

File: notifications.bs [hunk 1]
```diff
@@ -70,6 +70,11 @@ that no sounds or vibrations should be made.
 indicates that the screen of the device should not be enabled.
 
 <p>A <a lt="concept notification">notification</a> has an associated
+<dfn>require interaction preference flag</dfn> which is initially unset. When
+set, hints that the notification should remain readily available until the user
```
Comment (beverloo) [inline notifications.bs]: s/hints/indicates/? The rest of the sentences uses "... should remain ..." anyway.

**Comment by beverloo** [comment]
I'd like to get @sicking's opinion before we land this. It's very lightly defined, but I'm not sure how we can further define this without going in the realms of defining notification center behavior... :/

**Comment by annevk** [comment]
Getting @sicking's feedback would be good. I would also like to ask you to rebase this and merge the commits into a single commit (through `git push --force`). That would make it easier to merge later on.

File: notifications.bs (hunk 1 above)
Comment (johnmellor) [inline notifications.bs]: Done

**Comment by johnmellor** [comment]
@annevk wrote:
> I would also like to ask you to rebase this and merge the commits into a single commit

Done

File: notifications.bs [hunk 2]
```diff
@@ -70,6 +70,11 @@ that no sounds or vibrations should be made.
 indicates that the screen of the device should not be enabled.
 
 <p>A <a lt="concept notification">notification</a> has an associated
+<dfn>require interaction preference flag</dfn> which is initially unset. When
+set, indicates that the notification should remain readily available until the
```
Comment (annevk) [inline notifications.bs]: Perhaps clarify that it should only remain readily available if the screen is sufficiently large?

**Comment by annevk** [comment]
With that nit addressed and another rebase/force push I'll land this. Thank you!

File: notifications.bs (hunk 2 above)
Comment (johnmellor) [inline notifications.bs]: Done (and ran bikeshed/rebased/force pushed).

**Pull request merged**